lib/+ne7/+ip/imgaussian.m

```diff
@@ -0,0 +1,58 @@
+function I=imgaussian(I,sigma,siz)
+% IMGAUSSIAN filters an 1D, 2D color/greyscale or 3D image with an 
+% Gaussian filter. This function uses for filtering IMFILTER or if 
+% compiled the fast  mex code imgaussian.c . Instead of using a 
+% multidimensional gaussian kernel, it uses the fact that a Gaussian 
+% filter can be separated in 1D gaussian kernels.
+%
+% J=IMGAUSSIAN(I,SIGMA,SIZE)
+%
+% inputs,
+%   I: The 1D, 2D greyscale/color, or 3D input image with 
+%           data type Single or Double
+%   SIGMA: The sigma used for the Gaussian kernel
+%   SIZE: Kernel size (single value) (default: sigma*6)
+% 
+% outputs,
+%   J: The gaussian filtered image
+%
+% note, compile the code with: mex imgaussian.c -v
+%
+% example,
+%   I = im2double(imread('peppers.png'));
+%   figure, imshow(imgaussian(I,10));
+% 
+% Function is written by D.Kroon University of Twente (September 2009)
+
+if(~exist('siz','var')), siz=sigma*6; end
+
+if(sigma>0)
+    % Make 1D Gaussian kernel
+    x=-ceil(siz/2):ceil(siz/2);
+    H = exp(-(x.^2/(2*sigma^2)));
+    H = H/sum(H(:));
+
+    % Filter each dimension with the 1D Gaussian kernels\
+    if(ndims(I)==1)
+        I=imfilter(I,H, 'same' ,'replicate');
+    elseif(ndims(I)==2)
+        Hx=reshape(H,[length(H) 1]);
+        Hy=reshape(H,[1 length(H)]);
+        I=imfilter(imfilter(I,Hx, 'same' ,'replicate'),Hy, 'same' ,'replicate');
+    elseif(ndims(I)==3)
+        if(size(I,3)<4) % Detect if 3D or color image
```

Comment:
How do you know if it's a color image? If its 3rd dimension is 2 or 3?  A more reliable test or option should be used.